plugin/route53/route53.go

```diff
@@ -254,12 +254,55 @@ func (h *Route53) updateZones(ctx context.Context) error {
 				}
 				err = h.client.ListResourceRecordSetsPagesWithContext(ctx, in,
 					func(out *route53.ListResourceRecordSetsOutput, last bool) bool {
+						failoverList := map[string]map[string]*route53.ResourceRecordSet{}
 						for _, rrs := range out.ResourceRecordSets {
+							if rrs.SetIdentifier != nil {
+								if failoverList[*rrs.Name] == nil {
+									failoverList[*rrs.Name] = map[string]*route53.ResourceRecordSet{}
+								}
+								failoverList[*rrs.Name][*rrs.Failover] = rrs
+								continue
+							}
+
 							if err := updateZoneFromRRS(rrs, newZ); err != nil {
 								// Maybe unsupported record type. Log and carry on.
 								log.Warningf("Failed to process resource record set: %v", err)
 							}
 						}
+
+						for k := range failoverList {
```

Comment:
This should go into separate method and I don't think it's desired to have healtcheck performed within a hot loop - it should be performed async